src/functions/sendCandidateResults/application/service/categories/K/fault-provider-cat-k.ts

```diff
@@ -0,0 +1,57 @@
+import { CatKUniqueTypes } from '@dvsa/mes-test-schema/categories/K';
+import { Fault } from '../../../../domain/fault';
+import { CompetencyOutcome } from '../../../../domain/competency-outcome';
+import {
+  convertBooleanFaultObjectToArray,
+  convertNumericFaultObjectToArray, getVehicleCheckFaultCount,
+} from '../../fault-provider';
+import { Competencies } from '../../../../domain/competencies';
+
+export const getDrivingFaultsCatK = (testData: CatKUniqueTypes.TestData | undefined): Fault [] => {
+  const drivingFaults: Fault[] = [];
+
+  if (!testData) {
+    throw new Error('No Test Data');
+  }
+
+  if (testData.drivingFaults) {
+    convertNumericFaultObjectToArray(testData.drivingFaults)
+      .forEach(fault => drivingFaults.push(fault));
+  }
+
+  if (getVehicleCheckFaultCount(testData.vehicleChecks as CatKUniqueTypes.VehicleChecks, CompetencyOutcome.DF) >= 1) {
+    drivingFaults.push({ name: Competencies.vehicleChecks, count: 1 });
+  }
+
+  return drivingFaults;
+};
+
+export const getSeriousFaultsCatK = (testData: CatKUniqueTypes.TestData | undefined): Fault [] => {
+  const seriousFaults: Fault[] = [];
+
+  if (!testData) {
+    throw new Error('No Test Data');
+  }
+
+  if (testData.seriousFaults) {
+    convertBooleanFaultObjectToArray(testData.seriousFaults)
+      .forEach(fault => seriousFaults.push(fault));
+  }
+
+  return seriousFaults;
+};
+
+export const getDangerousFaultsCatK = (testData: CatKUniqueTypes.TestData | undefined): Fault [] => {
+  const dangerousFaults: Fault[] = [];
+
+  if (!testData) {
+    throw new Error('No Test Data');
+  }
+
+  if (testData.dangerousFaults) {
+    convertBooleanFaultObjectToArray(testData.dangerousFaults)
+      .forEach(fault => dangerousFaults.push(fault));
+  }
+
+  return dangerousFaults;
+};
```

Comment:
Don't like the idea of a fault provider for each home test category... pretty sure a common home-test provider could be created. Avoids duplication

Reply:
Done

Reply:
done